libshaderc/include/shaderc/shaderc.hpp

```diff
@@ -55,23 +69,23 @@ class CompilationResult {
     return shaderc_result_get_compilation_status(compilation_result_);
   }
 
-  // Returns a pointer to the start of the compiled SPIR-V.
-  // It is guaranteed that static_cast<uint32_t> is valid to call on this
-  // pointer.
-  // This pointer remains valid only until the CompilationResult is destroyed.
-  const char* GetData() const {
-    if (!compilation_result_) {
-      return "";
-    }
-    return shaderc_result_get_bytes(compilation_result_);
+  // Returns an iterator pointing to the start of the compilation
+  // output.  It is valid for the lifetime of this object.
+  // If there is no compilation result, then returns nullptr.
+  const_iterator cbegin() const {
```

Comment:
It is completely valid to also call these begin() and end() (and have them return const_iterators).
This way range-based for loops can/will still work correctly.